mobile/android/android-device-hash-service.ts

```diff
@@ -52,22 +52,32 @@ export class AndroidDeviceHashService implements Mobile.IAndroidDeviceHashServic
 		return false;
 	}
 
-	public async generateHashesFromLocalToDevicePaths(localToDevicePaths: Mobile.ILocalToDevicePathData[], shasums: IStringDictionary): Promise<string[]> {
+	public async generateHashesFromLocalToDevicePaths(localToDevicePaths: Mobile.ILocalToDevicePathData[], initialShasums: IStringDictionary = {}): Promise<IStringDictionary> {
```

Comment:
Why do we need to both update initialShasums and return the same object as a result?

Reply:
I prefer to implicitly return a result(more readable). However there is some code that depends on the shsums to be updated. I guess it can be accomplished only with return value, but I am not sure its worth the extra complication of the code.